content/en/docs/components/pipelines/v2/migration.md

```diff
@@ -444,6 +444,36 @@ def pipeline():
 </tr>
 </table>
 
+#### Typecasting behavior change
+
+**Affects:** KFP OSS users and Vertex AI Pipelines users
+
+KFP SDK v1 had more lenient pipeline typechecking than does KFP SDK v2. Some pipelines that utilized this leniency may not be compilable using KFP SDK v2. For example, parameters typed with `float` would accept the string `"0.1"`:
+
+```python
+from kfp.v2 import compiler
+from kfp.v2 import dsl
+from kfp import components
+
+
+@dsl.component
+def train(
+    number_of_epochs: int,
+    learning_rate: float,
+):
+    print(f"number_of_epochs={number_of_epochs}")
+    print(f"learning_rate={learning_rate}")
+
+
+def training_pipeline(number_of_epochs: int = 1):
+    train(
+        number_of_epochs=number_of_epochs,
+        learning_rate="0.1",
```

Comment:
nit: add an inline comment here to say this would be incompatible in v2, and what would be compatible in v2, so it's a little bit easier to the readers?